src/middlewared/middlewared/plugins/ipmi_/lan.py

```diff
@@ -118,7 +117,59 @@ def query(self, filters, options):
 
             result.append(data)
 
-        return filter_list(result, filters, options)
+        return result
+
+    @accepts(
+        Dict(
+            'ipmi_lan_query',
+            Ref('query-filters'),
+            Ref('query-options'),
+            Dict('ipmi_options', Bool('query_remote', default=False)),
+            register=True,
+        ),
+        roles=['IPMI_READ'],
+    )
+    @filterable_returns(
+        List(
+            'channels',
+            items=[
+                Dict(
+                    'channel',
+                    Int('channel'),
+                    Int('id'),
+                    Str('ip_address_source'),
+                    Str('ip_address'),
+                    Str('mac_address'),
+                    Str('subnet_mask'),
+                    Str('default_gateway_ip_address'),
+                    Str('default_gateway_mac_address'),
+                    Str('backup_gateway_ip_address'),
+                    Str('backup_gateway_mac_address'),
+                    Int('vlan_id'),
+                    Bool('vlan_id_enable'),
+                    Int('vlan_priority'),
+                )
+            ],
+        )
+    )
+    def query(self, data):
+        """Query available IPMI Channels with `query-filters` and `query-options`."""
+        if not data['ipmi_options']['query_remote']:
+            result = self.query_impl()
+        elif self.middleware.call_sync('failover.licensed'):
+            try:
+                result = self.middleware.call_sync(
+                    'failover.call_remote', 'ipmi.lan.query_impl'
+                )
+            except Exception:
+                # could be ENOMETHOD on upgrade or could be that
+                # remote node isn't connected/functioning etc OR
+                # could be that we're not on an HA system. In
+                # either of the scenarios, we just need to return
+                # an empty list
+                result = []
+
+        return filter_list(result, data['query_filters'], data['query_options'])
```

Comment:
If `data['ipmi_options']['query_remote']` is set and system is not licensed, `result` will be uninitialized ?